massa-protocol-worker-2/src/handlers/block_handler/propagation.rs

```diff
@@ -0,0 +1,179 @@
+                            {
+                                let cache_read = self.cache.read();
+                                for (peer_id, (blocks_known, _)) in &cache_read.blocks_known_by_peer
+                                {
+                                    // peer that isn't asking for that block
+                                    let cond = blocks_known.peek(&block_id);
+                                    // if we don't know if that peer knows that hash or if we know it doesn't
+                                    if !cond.map_or_else(|| false, |v| v.0) {
+                                        massa_trace!("protocol.protocol_worker.process_command.integrated_block.send_header", { "peer_id": peer_id, "block_id": block_id});
+                                        if let Some(connection) =
+                                            self.active_connections.read().connections.get(&peer_id)
+                                        {
+                                            if let Err(err) = connection.send_channels.send(
+                                                &self.block_serializer,
+                                                BlockMessage::BlockHeader(header.clone()).into(),
+                                                true,
+                                            ) {
+                                                warn!("Error while sending block header to peer {} err: {:?}", peer_id, err);
+                                            }
+                                        }
+                                    } else {
+                                        massa_trace!("protocol.protocol_worker.process_command.integrated_block.do_not_send", { "peer_id": peer_id, "block_id": block_id });
+                                    }
+                                }
+                            }
+                        }
+                        BlockHandlerCommand::AttackBlockDetected(block_id) => {
+                            //TODO: Ban all nodes that sent us this object
+                            let to_ban: Vec<PeerId> = self
+                                .cache
+                                .read()
+                                .blocks_known_by_peer
+                                .iter()
+                                .filter_map(|(id, (block_known, _))| {
+                                    match block_known.peek(&block_id) {
+                                        Some((true, _)) => Some(id.clone()),
+                                        _ => None,
+                                    }
+                                })
+                                .collect();
+                            for id in to_ban.iter() {
+                                massa_trace!("protocol.protocol_worker.process_command.attack_block_detected.ban_node", { "node": id, "block_id": block_id });
+                                let _ = self.ban_node(id);
```

Comment:
I think we should handle the error case, at least to display an error. Even more when it comes to banning dangerous peers

Reply:
Done